presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -190,6 +193,136 @@ public static Block bingTilesAround(
         return blockBuilder.build();
     }
 
+    @Description("Given a (longitude, latitude) point, returns the surrounding Bing tiles " +
+            "at the specified zoom level covering minimum radiusInKm")
+    @ScalarFunction("bing_tiles_around")
+    @SqlType("array(" + BingTileType.NAME + ")")
+    public static Block bingTilesAround(
+            @SqlType(StandardTypes.DOUBLE) double latitude,
+            @SqlType(StandardTypes.DOUBLE) double longitude,
+            @SqlType(StandardTypes.INTEGER) long zoomLevel,
+            @SqlType(StandardTypes.DOUBLE) double radiusInKm)
+    {
+        checkLatitude(latitude, LATITUDE_OUT_OF_RANGE);
+        checkLongitude(longitude, LONGITUDE_OUT_OF_RANGE);
+        checkZoomLevel(zoomLevel);
+
+        long mapSize = mapSize(toIntExact(zoomLevel));
+        long maxTileIndex = (mapSize / TILE_PIXELS) - 1;
+
+        long tileY = longitudeToTileY(latitude, mapSize);
+        long tileX = longitudeToTileX(longitude, mapSize);
+
+        // Find top, bottom, left and right tiles from center of circle
+        double[] topLatLong = GeoFunctions.addDistanceToLatitudeLongitude(latitude, longitude, radiusInKm, 0);
+        BingTile topTile = latitudeLongitudeToTile(correctLatitude(topLatLong[0]),
+                correctLongitude(topLatLong[1]),
+                toIntExact(zoomLevel));
+
+        double[] bottomLatLong = GeoFunctions.addDistanceToLatitudeLongitude(latitude, longitude, radiusInKm, 180);
```

Comment:
Does this code handle wrapping around the north and south polls and around 180th meridian correctly?

Reply:
Nope doesnt handle wrap around. Is that a requirement?